### PR TITLE
Add capability to check if service is successfully deployed

### PR DESCRIPTION
**Pull request by nikhilo**
Currently the module lacks support for updating existing service. Also there is no way to check whether service is successfully updated/deployed.
This adds the capability to update existing service with new values (except for ELB). It also checks with the `repeat` and `delay` parameters (similar to delete) whether the service is running desired number of tasks or not.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @nikhilo. @Java1Guy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by Java1Guy** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @nikhilo for this PR, and thanks @Java1Guy for reviewing. Marking for inclusion.

**Comment by nikhilo** [comment]
:+1:

**Comment by nikhilo** [comment]
I have encountered a weird condition where a deployment fails due to unavailable resources, but the module does not detect/report it. I will reproduce that, and will come up with a better solution on checking the service deployment.

**Comment by nikhilo** [comment]
I couldn't figure out what this meant: `ERROR: version_added for new option (follow) should be 2.1. Currently 0.0`

**Comment by resmo** [comment]
Thanks @nikhilo for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by nikhilo** [comment]
ready_for_review

File: cloud/amazon/ecs_service.py [hunk 1]
```diff
@@ -416,6 +425,34 @@ def main():
             module.fail_json(msg="Service still not deleted after "+str(repeat)+" tries of "+str(delay)+" seconds each.")
             return
 
+    elif module.params['state'] == 'deployed':
+        if not existing:
+            module.fail_json(msg="Service '"+module.params['name']+" not found.")
+            return
```
Comment (resmo) [inline cloud/amazon/ecs_service.py]: no need for a return here

File: cloud/amazon/ecs_service.py [hunk 2]
```diff
@@ -416,6 +425,34 @@ def main():
...
+                    break
+            if status == "ACTIVE" and desired_count == running_count:
+                results['changed'] = True
+                results['service'] = service_mgr.jsonize(existing)
+                successful = True
+                break
+            time.sleep(delay)
+        if not successful:
+            module.fail_json(msg="Service still not deployed after "+str(repeat)+" tries of "+str(delay)+" seconds each. Failure message: "+existing['events'][0]['message'])
+            return
```
Comment (resmo) [inline cloud/amazon/ecs_service.py]: no need for a return here

**Comment by resmo** [comment]
Thanks @nikhilo. To the current maintainers, @Java1Guy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: cloud/amazon/ecs_service.py (hunk 2 above)
Comment (Java1Guy) [inline cloud/amazon/ecs_service.py]: seems like that's the pattern... return after calling fail_json.

**Comment by nikhilo** [comment]
@resmo you're right, no return needed after `fail_json`. Removed returns

**Comment by resmo** [comment]
@nikhilo does the new state `deployed` also handle the check mode? Because the module says it has check mode support and I didn't find anything about check mode in the new code.

**Comment by nikhilo** [comment]
@resmo Thanks for reminding that. Let me observe check_mode for 'deployed' state.
Also, the 'extra return' and 'check_mode' problems are in the 'deleting' state as well. Should these be addressed here? or in another PR?

**Comment by resmo** [comment]
@nikhilo make a separate PR but I would wait until this PR has been merged. Otherwise you may have to rebase again

**Comment by nikhilo** [comment]
Of course. Thanks @resmo

**Comment by gregdek** [comment]
Thanks again to @nikhilo for this PR, and thanks @Java1Guy for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by akvadrako** [comment]
I don't see in the diff where this changes anything about updating the service.  Has that part been removed or already incorporated?

**Comment by nikhilo** [comment]
@akvadrako yes, update service was added in #1726

**Pull request closed without merging**